### ISSUE TITLE
[BUG] API: Roles specified when creating user are not mapped to that user

### ISSUE THREAD

**Issue by philvarner** [opensearch-project/security#2345]
**What is the bug?**

When a user is created via the api with roles assigned, the role is not mapped to the user.

**How can one reproduce the bug?**
1. Create a Role named `my_role`

2. curl the API to create a new user `kirk` with that role applied:

```
curl -X "PUT" "${HOST}/_plugins/_security/api/internalusers/kirk" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -u 'admin:xxxx' \
     -d $'{
  "opendistro_security_roles": [
    "my_role"
  ],
  "password": "xxxx1@aA"
}'
```

3. The user `kirk` is created, but the role `my_role` is not mapped to `kirk`

**What is the expected behavior?**

The user `kirk` is created and the role `my_role` is mapped to `kirk`.

**What is your host/environment?**

AWS OpenSearch Service, OpenSearch 2.3

**Do you have any screenshots?**

n/a

**Do you have any additional context?**

n/a

**Comment by DarshitChanpura** [comment]
[TRIAGE] @opendistroforelasticsearch-security Can someone please look into this?

**Comment by peternied** [comment]
Unassigned myself, I wasn't getting around to working on this issue.

**Comment by jiten1551** [comment]
I don't think that issue with API it's more on documentation where creating user with `opendistro_security_roles`  gives user permission with mentioned roles let's say` all_access` but ` GET _plugins/_security/api/rolesmapping` will not show this mapping as it's only meant for explicit mapping
